crates/solidity/testing/sanctuary/src/main.rs

```diff
@@ -16,22 +18,25 @@ use crate::{
     reporting::Reporter,
 };
 
-fn main() {
+fn main() -> Result<ExitCode> {
     let versions = SolidityDefinition::create().collect_breaking_versions();
 
-    // Fail the parent process if a child thread panics:
-    std::panic::catch_unwind(|| -> Result<()> {
-        for dataset in get_all_datasets()? {
-            process_dataset(&dataset, &versions)?;
+    for dataset in get_all_datasets()? {
+        let res = catch_unwind(|| process_dataset(&dataset, &versions)).expect("A child panicked.");
```

Comment:
Nit: keeping the comment:

```suggestion
        // Fail the parent process if a child thread panics:
        let res = catch_unwind(|| process_dataset(&dataset, &versions)).expect("A child panicked.");
```